core/org.wso2.carbon.core/src/main/java/org/wso2/carbon/core/util/KeyStoreManager.java

```diff
@@ -118,99 +127,91 @@ public static KeyStoreManager getInstance(int tenantId,
-            org.wso2.carbon.registry.api.Resource resource;
-            KeyStore keyStore;
-
-            if (registry.resourceExists(path)) {
-                resource = registry.get(path);
-            } else {
-                throw new SecurityException("Given Key store is not available in registry : " + keyStoreName);
+            if (KeyStoreUtil.isCustomKeyStore(keyStoreName)) {
+                return getCustomKeyStorePrivateKey(keyStoreName);
             }
 
-            CryptoUtil cryptoUtil = CryptoUtil.getDefaultCryptoUtil();
-            String encryptedPassword = resource
-                    .getProperty(RegistryResources.SecurityManagement.PROP_PRIVATE_KEY_PASS);
-            String privateKeyPasswd = new String(cryptoUtil.base64DecodeAndDecrypt(encryptedPassword));
-
-            if (isCachedKeyStoreValid(keyStoreName)) {
-                keyStore = loadedKeyStores.get(keyStoreName).getKeyStore();
-                return keyStore.getKey(alias, privateKeyPasswd.toCharArray());
-            } else {
-                byte[] bytes = (byte[]) resource.getContent();
-                String keyStorePassword = new String(cryptoUtil.base64DecodeAndDecrypt(resource.getProperty(
-                        RegistryResources.SecurityManagement.PROP_PASSWORD)));
-                keyStore = KeyStore.getInstance(resource
-                        .getProperty(RegistryResources.SecurityManagement.PROP_TYPE));
-                ByteArrayInputStream stream = new ByteArrayInputStream(bytes);
-                keyStore.load(stream, keyStorePassword.toCharArray());
-
-                KeyStoreBean keyStoreBean = new KeyStoreBean(keyStore, resource.getLastModified());
-                updateKeyStoreCache(keyStoreName, keyStoreBean);
-                return keyStore.getKey(alias, privateKeyPasswd.toCharArray());
-            }
+            return getTenantPrivateKey(keyStoreName, alias);
         } catch (Exception e) {
             log.error("Error loading the private key from the key store : " + keyStoreName);
             throw new SecurityException("Error loading the private key from the key store : " +
                     keyStoreName, e);
         }
     }
 
+    /**
+     * This method loads the public certificate of a given key store.
+     *
+     * @param keyStoreName  Name of the key store.
+     * @param alias         Alias of the certificate.
+     * @return Public Certificate corresponding to the alias.
+     * @throws Exception    If there is an error when retriving the public certificate from given keystore.
+     */
+    public Certificate getCertificate(String keyStoreName, String alias) throws Exception {
+
+        if (keyStoreName == null || keyStoreName.isEmpty()) {
+            throw new SecurityException("Key Store Name is null or empty.");
+        }
+        if (alias == null || alias.isEmpty()) {
+            throw new SecurityException("Alias is null or empty.");
```

Comment:
```suggestion
            throw new SecurityException("Certificate alias is null or empty.");
```

Reply:
FIxed in https://github.com/wso2/carbon-kernel/pull/4025/commits/3aa02c8fd1024563e7aefe9329049af4802dfac3